Content.Server/_DV/CosmicCult/EntitySystems/CosmicColossusSystem.cs

```diff
@@ -0,0 +1,139 @@
+        SubscribeLocalEvent<CosmicColossusComponent, EventCosmicColossusIngressDoAfter>(OnColossusIngressDoAfter);
+        SubscribeLocalEvent<CosmicColossusComponent, EventCosmicColossusSunder>(OnColossusSunder);
+
+    }
+
+    private void OnMobStateChanged(Entity<CosmicColossusComponent> ent, ref MobStateChangedEvent args)
+    {
+        if (args.NewMobState is not MobState.Alive)
+        {
+            if (!TryComp<PhysicsComponent>(ent, out var physComp))
+                return;
+            _appearance.SetData(ent, ColossusVisuals.Status, ColossusStatus.Dead);
+            _ambientSound.SetAmbience(ent, false);
+            _audio.PlayPvs(ent.Comp.DeathSFX, ent);
+            _physics.SetBodyStatus(ent, physComp, BodyStatus.OnGround, true);
+            _popup.PopupCoordinates(
+                Loc.GetString("cosmiccult-colossus-death"),
+                Transform(ent).Coordinates,
+                PopupType.Large);
+            RemComp<PointLightComponent>(ent);
+        }
+    }
+
+    private void OnColossusIngress(Entity<CosmicColossusComponent> ent, ref EventCosmicColossusIngress args)
+    {
+        var doargs = new DoAfterArgs(EntityManager, ent, ent.Comp.IngressDoAfter, new EventCosmicColossusIngressDoAfter(), ent, args.Target)
+        {
+            DistanceThreshold = 2f,
+            Hidden = false,
+            BreakOnMove = true,
+        };
+        args.Handled = true;
+        _audio.PlayPvs(ent.Comp.DoAfterSFX, ent);
+        _doAfter.TryStartDoAfter(doargs);
+    }
+
+    private void OnColossusIngressDoAfter(Entity<CosmicColossusComponent> ent, ref EventCosmicColossusIngressDoAfter args)
+    {
+        if (args.Args.Target is not { } target)
+            return;
+        if (args.Cancelled || args.Handled)
+            return;
+        args.Handled = true;
+        var comp = ent.Comp;
+
+        if (TryComp<DoorBoltComponent>(target, out var doorBolt))
+            _door.SetBoltsDown((target, doorBolt), false);
+        _door.StartOpening(target);
+        _audio.PlayPvs(comp.IngressSFX, ent);
+        Spawn(comp.CultVFX, Transform(target).Coordinates);
+    }
+
+    private void OnColossusSunder(Entity<CosmicColossusComponent> ent, ref EventCosmicColossusSunder args)
+    {
+        _appearance.SetData(ent, ColossusVisuals.Status, ColossusStatus.Attacking);
+        _transform.SetCoordinates(ent, args.Target);
+        _transform.AnchorEntity(ent);
+        _audio.PlayPvs(ent.Comp.TileSFX, ent);
+
+        args.Handled = true;
+
+        Spawn("CosmicColossusAttack1Vfx", args.Target);
+        Timer.Spawn(ent.Comp.ReleaseDelay, () => { _appearance.SetData(ent, ColossusVisuals.Status, ColossusStatus.Alive); _transform.Unanchor(ent); });
+
+        var area = Spawn(null, _transform.ToMapCoordinates(args.Target), null, default);
+        for (var size = 1; size <= 5; size++)
+        {
+            var range = size;
+            Timer.Spawn(TimeSpan.FromSeconds(size * 0.5), () => { DetonateTiles(area, range); });
```

Comment:
store the tile coords when using the ability instead of spawning entities just to get their positions